include/alpaka/api/host/atomic.hpp

```diff
@@ -163,10 +163,10 @@ namespace alpaka::onAcc
                 isSupportedByAtomicAtomicRef<T>();
                 detail::atomic_ref<T> ref(*addr);
                 T old = ref;
-                T result = ((old >= value) ? 0 : static_cast<T>(old - 1));
+                T result = (old == static_cast<T>(0) || old > value) ? value : (old - static_cast<T>(1));
                 while(!ref.compare_exchange_weak(old, result, detail::memory_order_relaxed))
                 {
-                    result = ((old >= value) ? 0 : static_cast<T>(old - 1));
+                    result = (old == static_cast<T>(0) || old > value) ? value : (old - static_cast<T>(1));
                 }
```

Comment:
This can be replaced with a do while, and the casts simply be direct initializations
```suggestion
                T result;
                do {
                    result = (old == T{0} || old > value) ? value : (old - T{1});
                } while (!ref.compare_exchange_weak(old, result, detail::memory_order_relaxed));
```

Reply:
I applied the suggestion to INc and Dec because in Inc there was a casting issue. This change can also be used for many other functions, but I do not want to sneak it into this PR